packages/react-graphql/src/hooks/query.ts

```diff
@@ -184,9 +183,7 @@ export default function useQuery<
       networkStatus: result.networkStatus,
       loading: result.loading,
     };
-
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [responseId, skip, queryObservable, defaultResult, previousData]);
+  }, [isLoading, skip, queryObservable, defaultResult, previousData]);
```

Comment:
Using the responseId which was just a counter was causing and extra repaint. With this we still subscribe to the observer (which gives us the query when its loaded) so we can use this to bust the cache and get the correct amount of expected calls.